app/views/posts/_posts.html.erb

```diff
@@ -0,0 +1,39 @@
+<% @user_posts.each do |user_post| %>
+    <div class="p-3 mb-3 bg-secondary rounded col-md-8 offset-md-2 text-white">
+      <small>Author: <%=  user_post.user.first_name %> <%=  user_post.user.last_name %></small><br>
+      <%= user_post.post_text %>
+      <% if current_user.id == user_post.user.id %>
+          <%= button_to 'delete', user_post, method: :delete, data: { confirm: 'Are you sure?' }, class:'btn btn-danger btn-sm' %> 
+      <% end %>
+
+       <%= form_for @like do |f| %>
+        <%= hidden_field_tag "user_id", "#{current_user.id}" %>
+        <%= hidden_field_tag "post_id", "#{user_post.id}" %>
+        <div class="actions">
+            <%= f.submit "Like", class: "form-control btn btn-success mt-1" %>
+        </div>
+       <% end %>
+      
+      <%= user_post.likes.size %>
+      
+      <%= form_for [@comment] do |f| %>
+        <%= hidden_field_tag "post_id", "#{user_post.id}" %>
+        <div class="field">
+            <%= f.label 'Comment' %><br />
+            <%= f.text_area :comment_text, class: "form-control", placeholder: "Comment on this post...", id:"comment-#{user_post.id}" %>
+        </div>
+        <div class="actions">
+            <%= f.submit "Create Comment", class: "form-control btn btn-success mt-1", id: "button-#{user_post.id}" %>
+        </div>
+      <% end %>
+    </div>
+    <% if !user_post.comments.empty? %>
```

Comment:
- [ ] Same here, no logic in the views, but in this case you will find that logic that should move into a helper is too much. So I would suggest that you create a partial for comments, then render it using [collection rendering](https://guides.rubyonrails.org/layouts_and_rendering.html#rendering-collections) inside the helper method.